torchvision/prototype/transforms/_container.py

```diff
@@ -1,19 +1,20 @@
-from typing import Any, Dict, List, Optional
+from typing import Any, Callable, Dict, List, Optional, Sequence
 
 import torch
 from torchvision.prototype.transforms import Transform
+from torchvision.utils import _log_api_usage_once
 
 from ._transform import _RandomApplyTransform
 
 
-class Compose(Transform):
-    def __init__(self, *transforms: Transform) -> None:
-        super().__init__()
+class Compose:
```

Comment:
Can we at least define this as an `nn.Module` and structure it with a forward?

Reply:
Yes, we can inherit from nn.Module without registering `transforms` as submodules otherwise passing just a callable wont work

Reply:
@pmeier Thoughts on this?

Reply:
`Transform` is already thin wrapper around `nn.Module` and if we overwrite the `forward` method without calling `super().forward` the only difference is the added `__repr__` as well as logging. Thus, if we can derive from `nn.Module`, we should be able to derive from `Transform`.

Reply:
I also agree that deriving from `Transform` for all classes is preferable. The less exceptions the better IMO.

Reply:
What is the benefit of deriving `Transform` from `nn.Module` ?

Reply:
For me it's weird we propose a new API and `Compose` is this weird thing that doesn't follow it. IMO inheriting from `Transform` and overwriting `forward()` is something perfectly aligned with the new API. It's something we did discuss as possibility during designs.

From the comments in this PR I think it might be more beneficial to sync with Philip offline and align. I can join you later if that's needed and we can summarize the decision here afterwards.